Lib/asyncio/runners.py

```diff
@@ -176,8 +176,12 @@ async def main():
         raise RuntimeError(
             "asyncio.run() cannot be called from a running event loop")
 
-    with Runner(debug=debug) as runner:
-        return runner.run(main)
+    try:
+        with Runner(debug=debug) as runner:
+            events.set_event_loop(runner.get_loop())
+            return runner.run(main)
+    finally:
+        events.set_event_loop(None)
```

Comment:
In python 3.10 this ran before the loop is closed, so that when ChildWatcher.attach_loop gets called the old loop is still open